src/main/java/io/sqooba/traildb/TrailDBIterator.java

```diff
@@ -52,6 +52,9 @@ public TrailDBEvent next() {
             public boolean hasNext() {
                 TrailDBEvent next = TrailDBNative.INSTANCE.cursorNext(TrailDBIterator.this.cursor);
                 if (next == null) {
+                    // Auto-close when at the end because it is not intuitive to have to close an
+                    // iterator when at its end in Java.
+                    TrailDBIterator.this.close();
```

Comment:
Ok, I would still rephrase it a bit... something like "Auto-close the iterator when it reaches its end. Do this explicitly because, in Java, it is not intuitive to close an iterator that reaches its end".
And indeed this auto-close is not intuitive to me :) Shouldn't you do it on the next() method as well then ? 
But I also see that the TrailDBIterator implements AutoCloseable. Couldn't we force (or at least encourage) the users to use it with a try-with-resources statement ? The users should know how to use the structure they use. If it implements autocloseable, I would not force a close when the iterator reaches its end.